DevelopGitHub/GitHubExtension/GitHubExtension.Security.Tests/TestForControllers/AccountControllerAssignRolesToUserTests.cs

```diff
@@ -0,0 +1,127 @@
+            User userToUpdate = new User
+            {
+                Id = "111",
+                ProviderId = 1,
+                UserRepositoryRoles = new List<UserRepositoryRole>
+               {
+                   new UserRepositoryRole { Id =2 , },
+                   new UserRepositoryRole { Id =3 , },
+               }
+            };
+            List<User> users = new List<User>
+            {
+                new User{ProviderId = 4,},
+                new User{ProviderId = providerId},
+                userToUpdate,
+            };
+            IEnumerable<SecurityRole> roles = new List<SecurityRole>
+            {
+                new SecurityRole{ Id = 1, Name = "Admin" },
+                new SecurityRole{ Id = 2, Name = "Developer" },
+                new SecurityRole{ Id = 3, Name = "Reviewer" }
+            };
+            UserRepositoryRole repositoryRoleToFind = new UserRepositoryRole { Id = 12 };
+            var userManager = Substitute.For<ApplicationUserManager>(Substitute.For<IUserStore<User>>());
+            userManager.Users.Returns(new MockForEnumerableQuery<User>(users));
+            userManager.UpdateAsync(userToUpdate).Returns(IdentityResult.Success);
+            userManager.CreateIdentityAsync(userToUpdate, DefaultAuthenticationTypes.ApplicationCookie).Returns(Task.FromResult(new ClaimsIdentity()));
+            userManager.AddClaimAsync(userToUpdate.Id, Arg.Any<Claim>()).Returns(IdentityResult.Success);
+            var context = Substitute.For<ISecurityContext>();
+            context.SecurityRoles.Returns(new MockForDbSet<SecurityRole>(roles));
+
+            return new AccountController(Substitute.For<IGithubService>(),
+                context, userManager);
+        }
+
+        [Theory]
+        [MemberData("GetDataForNotFountResult")]
+        public void NotFoundUserTest(AccountController controller, int gitHubId, int repoId, string roleToAssign)
+        {
+            Task<IHttpActionResult> response = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign);
+
+            IHttpActionResult result = response.Result;
+            result.Should().BeOfType<NotFoundResult>("Because user with providerId= {0} doesn't exists in database",gitHubId);
+        }
+
+        [Theory]
+        [MemberData("GetDataForInvalidModelStateResult")]
+        public void InvalidRoleTest(AccountController controller, int gitHubId, int repoId, string roleToAssign)
+        {
+            Task<IHttpActionResult> response = controller.AssignRolesToUser(repoId, gitHubId, roleToAssign);
+
+            IHttpActionResult result = response.Result;
+            result.Should().BeOfType<InvalidModelStateResult>("Because impossible to assign role = {0} that doesn't exist in database", roleToAssign);
+        }
+
+        [Theory]
+        [MemberData("GetDataForInvalidModelStateResult")]
+        public void ErrorMessageForInvalidRoleTest(AccountController controller, int gitHubId, int repoId, string roleToAssign)
```

Comment:
Please do't mix configuration entity, in our case controller, with arrange phase for test data. It should be done separately.